### PR TITLE
keep reference to repo in GitDiffStats

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
👍 good call

**Review by kshyatt — APPROVED** [review]

**Comment by ararslan** [comment]
Needs conflicts resolved

**Comment by simonbyrne** [comment]
I'm not going to be able to to do this for a day or 2, so if someone else
wants to update it, go ahead.

**Comment by ararslan** [comment]
Sorry, forgot I could do that myself! Conflicts have been resolved.

File: base/libgit2/diff.jl [hunk 1]
```diff
@@ -39,7 +39,7 @@ function GitDiffStats(diff::GitDiff)
     @check ccall((:git_diff_get_stats, :libgit2), Cint,
                   (Ptr{Ptr{Void}}, Ptr{Void}),
                   diff_stat_ptr_ptr, diff.ptr)
-    return GitDiffStats(diff_stat_ptr_ptr[])
+    return GitDiffStats(diff.repo, diff_stat_ptr_ptr[])
```
Comment (simonbyrne) [inline base/libgit2/diff.jl]: This should now be diff.owner

Comment (ararslan) [inline base/libgit2/diff.jl]: Fixed

**Pull request merged**